src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetFilter.cs

```diff
@@ -22,23 +22,33 @@ private ConventionBasedEntitySetFilter(Type targetType)
             this.targetType = targetType;
         }
 
+        public IQueryExpressionFilter Inner { get; set; }
+
         /// <inheritdoc/>
         public static void ApplyTo(
             IServiceCollection services,
             Type targetType)
         {
             Ensure.NotNull(services, "services");
             Ensure.NotNull(targetType, "targetType");
-            services.CutoffPrevious<IQueryExpressionFilter>(new ConventionBasedEntitySetFilter(targetType));
+            services.ChainPrevious<IQueryExpressionFilter>(next => new ConventionBasedEntitySetFilter(targetType)
+            {
+                Inner = next,
+            });
         }
 
         /// <inheritdoc/>
         public Expression Filter(QueryExpressionContext context)
         {
             Ensure.NotNull(context, "context");
-            if (context.ModelReference == null)
+
+            if (Inner != null)
             {
-                return null;
+                var innerFilteredExpression = Inner.Filter(context);
+                if (innerFilteredExpression != null)
```

Comment:
Typically a visitor returns the original node if no change happens, which allows us to give different meaning for a "null" value (like we want to abandon this node). The behavior of `Filter` here looks more like a mapping function thus IMO returning the original makes more sense.